src/web5-connect/sql-ttl-cache.ts

```diff
@@ -0,0 +1,126 @@
+import type { Dialect } from '@tbd54566975/dwn-sql-store';
+import { Kysely } from 'kysely';
+
+/**
+ * The SqlTtlCache is responsible for storing and retrieving cache data with TTL (Time-to-Live).
+ */
+export class SqlTtlCache {
+  private static readonly cacheTableName = 'cacheEntries';
+  private static readonly cleanupIntervalInSeconds = 60;
+
+  private db: Kysely<CacheDatabase>;
+  private cleanupTimer: NodeJS.Timeout;
+
+  private constructor(sqlDialect: Dialect) {
+    this.db = new Kysely<CacheDatabase>({ dialect: sqlDialect });
+  }
+
+  /**
+   * Creates a new SqlTtlCache instance.
+   */
+  public static async create(sqlDialect: Dialect): Promise<SqlTtlCache> {
+    const cacheManager = new SqlTtlCache(sqlDialect);
+
+    await cacheManager.initialize();
+
+    return cacheManager;
+  }
+
+  private async initialize(): Promise<void> {
+    await this.db.schema
+      .createTable(SqlTtlCache.cacheTableName)
+      .ifNotExists()
+      // 512 chars to accommodate potentially large `state` in Web5 Connect flow
+      .addColumn('key', 'varchar(512)', (column) => column.primaryKey())
+      .addColumn('value', 'text', (column) => column.notNull())
+      .addColumn('expiry', 'integer', (column) => column.notNull())
+      .execute();
+
+    // Start the cleanup timer
+    this.startCleanupTimer();
+  }
+
+  /**
+   * Starts a timer to periodically clean up expired cache entries.
+   */
+  private startCleanupTimer(): void {
+    this.cleanupTimer = setInterval(async () => {
+      await this.cleanUpExpiredEntries();
+    }, SqlTtlCache.cleanupIntervalInSeconds * 1000);
+  }
+
+  /**
+   * Inserts a cache entry.
+   * @param ttl The time-to-live in seconds.
+   */
+  public async insert(key: string, value: object, ttl: number): Promise<void> {
+    const expiry = Date.now() + (ttl * 1000);
+
+    const objectString = JSON.stringify(value);
+
+    await this.db
+      .insertInto(SqlTtlCache.cacheTableName)
+      .values({ key, value: objectString, expiry })
+      .execute();
+  }
+
+  /**
+   * Retrieves a cache entry if it is not expired and cleans up expired entries.
+   */
+  public async get(key: string): Promise<object | undefined> {
+    // clean up expired entries but no need to await for it to finish
+    this.cleanUpExpiredEntries();
```

Comment:
sanity: What is the purpose of calling this here since it's running on an interval + the particular record's expiry is checked/deleted below?

Reply:
Good point, keeping it simple. Removed.